### ISSUE TITLE
Warning in IDE module: Redundant superinterface IPartialEditingContentAssistParser

### ISSUE THREAD

**Issue by miklossy** [eclipse/xtext-core#1418]
When creating a new `MyDsl` Xtext project and running the `GenerateMyDsl.mwe2` workflow, Xtext generates the following code in the `org.xtext.example.mydsl.ide` module:
```
public class PartialMyDslContentAssistParser extends MyDslParser implements IPartialEditingContentAssistParser {
   ...
}
```
The `implements IPartialEditingContentAssistParser` part is not necessary, since the `AbstractContentAssistParser` already implements the `IPartialEditingContentAssistParser` interface.

That's why the following warning is reported by JDT:
![image](https://user-images.githubusercontent.com/20393472/76161771-c20d4800-6136-11ea-980e-0a44d25248bc.png)

**Hint:** The severity of this issue is `Ignore` by default, the warning will be reported after switching the severity of the `Redundant super interface` issue to `Warning`:
![image](https://user-images.githubusercontent.com/20393472/76161805-16b0c300-6137-11ea-8ad5-e82fe360eb61.png)

I encountered this problem while importing the [Xtext Seven Languages](https://github.com/xtext/seven-languages-xtext) projects into the workspace, where the severity of that issue is already configured to `Warning`:
![image](https://user-images.githubusercontent.com/20393472/76161878-c8e88a80-6137-11ea-9c6a-c98d06eeced0.png)

**Comment by cdietrich** [comment]
in wonder why we have that duplicate interface and if there are any cases its actually needed.

**Comment by cdietrich** [comment]
Fixed in 2.22
